dev/com.ibm.ws.jpa.tests.jpa_32_fat/test-applications/jakartadata/src/io/openliberty/jpa/data/tests/web/JakartaDataRecreateServlet.java

```diff
@@ -562,6 +562,40 @@ public void testOLGH28925() throws Exception {
 
         assertEquals(4, primes.size());
     }
+    @Test
+    // "Reference issue: https://github.com/OpenLiberty/open-liberty/issues/30093"
+    public void testOLGH30093() throws Exception {
+        deleteAllEntities(Prime.class); // Cleanup any left over entities
+
+        Prime two = Prime.of(2, "II", "two");
+        Prime three = Prime.of(3, "III", "three");
+        Prime five = Prime.of(5, "V", "five");
+        Prime seven = Prime.of(7, "VII", "seven");
+
+        List<Prime> primes;
+
+        tx.begin();
+        em.persist(two);
+        em.persist(three);
+        em.persist(five);
+        em.persist(seven);
+        tx.commit();
+
+        tx.begin();
+        try {
+            primes = em.createQuery(
+                                    "SELECT ID(THIS) FROM Prime WHERE ID(THIS) < ?1 ORDER BY ID(THIS) DESC",
+                                    Prime.class)
+                            .setParameter(1, 7)
+                            .getResultList();
+            tx.commit();
+        } catch (Exception e) {
+            tx.rollback();
+            throw e;
+        }
+
+        assertEquals(3, primes.size());
```

Comment:
```suggestion
        assertEquals(3, ids.size());
        assertEquals(5L, ids.get(0));
        assertEquals(3L, ids.get(1));
        assertEquals(2L, ids.get(2));
```

Since your query has an `ORDER BY ID(THIS) DESC` clause, you should also assert that the order of elements returned is correct.

Reply:
Hi @KyleAure I have made changes as you mentioned .